### ISSUE TITLE
NPC Summons Despawning

### ISSUE THREAD

**Issue by ccshiro** [ccshiro/cc-buglist#1058]
Currently minions summoned in combat despawn when the owner dies. This is not fully correct. For example, some summoned minions by NPCs should despawn as the master dies, others should not despawn unless you make them evade (usually after they stand around a bit first), etc.

We would like to fix it so it works correctly for the various NPC owned minions. However, that requires us gathering a fairly decent knowledge base of the various NPC minions and their despawn behavior. Before we have that, we can't really commit to one method of fixing it.

In short, we're looking for NPCs that spawn minions, and how it works. I will provide a made up example of the type of data we need:

```
NPC: Affronted Summoner
Minion: Smelly bent-wing bat
Summon: Summoner summons the bat a few seconds into combat.
Despawn: If the summoner evades, bat despawns.
         Also, if bat evades, it despawns after a few seconds.
```

In short we'd like to compile a list of NPCs with this type of data.

**Comment by r00ty-tc** [comment]
I was wondering if either spell.dbc or summonproperties.dbc had some indicator to show which should despawn and which should remain.

I suppose in order to find out though we need at least a few examples of each to find a common denominator (if there is even such a thing).

**Comment by Ghhlox** [comment]
No proof, just my memory. But perhaps it will help someone narrow their searches.

A good rule of thumb is that if the summon was an instant cast (except for special things like the mines in Gnomer or Blood Furnace) then it would despawn once the summoner was slain but channeled summons would have to be slain even after the summoner was dead. That should cover most situations, and then the special cases can be filled in later.

**Comment by DeeMarie** [comment]
Kraator (rare SMV mob) video from 2007, showing _channeled_ summoned infernals, that do not despawn on master death (or give xp):

https://www.youtube.com/watch?v=9GfymZjRwJY

And one of Baelmon the Houndmaster doing a near instant summon on a dog that appears to despawn (the players are instantly ooc when he dies)

https://youtu.be/H97dScZ581U?t=293

**Comment by ConfusedCC** [comment]
in the Baelmon the Houndmaster video you can see that there is a edit in the video at the 6:06 mark, notice the change in buffs, and the looting of a seccond mob(this might be an accidental pull or respawn instead of the summoned mob but thats hard to tell due to the video edit)

**Comment by YourNewfriend** [comment]
https://github.com/ccshiro/cc-buglist/issues/73

**Comment by ConfusedCC** [comment]
NPC: Wrath Priestess
Minion: Fel Imp
Summon: Summoner summons the Fel Imp on initial spawn
Despawn: Doesnt despawn when wrath priestess dies(https://youtu.be/cMJzSmJSPGY?t=2m15s) video shows the end of a multipull where the fel imp is the last mob alive, he gets banished so the warlock can heal his daemon

**Comment by ConfusedCC** [comment]
NPC: grishna Falconwing 
Minion: Grisna Raven
Summon: Summoner summons the Grisna Raven on initial spawn
Despawn: Doesnt despawn when owner dies
Proof
(https://youtu.be/HU1i355wPT8?t=1m36s) video unfortunately from 2014, earliest relevant footage found

comment #248 at http://wowdata.buffed.de/quest/Schergrat/Das-Fluestern-des-Rabengottes-10607 
states that he killed the grishna falconwing and got the buff (to understand ravenspeach) and that the raven was still alive
![raven still alive after master dies](https://cloud.githubusercontent.com/assets/10581981/9310771/4013cd50-4512-11e5-8f95-ad4f810b5f84.JPG)

**Comment by ConfusedCC** [comment]
NPC: Shadowmoon Summoner
Minion: Seductress
Summon: Summoner summons the seductress after being aggroed, often casts a 
Fireball before the summon
Despawn: Doesnt despawn when owner dies
Proof
http://www.gametrailers.com/videos/6g0muo/wow-bc-the-blood-furnace time: 03:12
Note: on the site the posted date is 2011, but the comments to the video is 8 years old.
the video seems to have been made at the tbc test realm before the release.

**Comment by YourNewfriend** [comment]
NPC: King Bangalash
Minion: Jungle Panthers
Summon: Summons them occasionally during combat
Despawn: They do not despawn
Evidence: https://youtu.be/EWlLM3dlNZw?t=3m8s

**Comment by Chulain** [comment]
http://www.wowhead.com/npc=19657/summoner-kanthin#comments
-1
 2007/04/19 (Patch 2.0.12)
The elemental now despawns when you defeat him.

## 

3
snowgoon on 2007/04/19 (Patch 2.0.12)
it didn't despawn for me

## 

12
doown on 2007/06/01 (Patch 2.1.0)
seems that if you don't damage the elemental, it despawns at Kanthin's death.

## 

0
warlow on 2008/09/30 (Patch 2.4.3)
The elemental did not despawn for me even though I killed Kanthin without dmging the elemental

## 

0
Chrome166 on 2010/08/10 (Patch 3.3.5)
Pets despawn if you kill their owners from range in one hit. In other words, kill the owner without causing the pet to aggro.

**Comment by KimmoKM** [comment]
A [Wowhead comment](http://www.wowhead.com/npc=17308/omor-the-unscarred#comments:id=42937) suggests the fel hunter adds at Omor the Unscarred had to be deal with after the boss died. Currently, they despawn upon his death.

**Comment by YourNewfriend** [comment]
Yea, at this point, we need to look for something that actually did despawn at its master's death on retail.

**Comment by KimmoKM** [comment]
At this point it indeed might be more productive to look for minions that DO despawn but here's another one:

The imps summoned by [Mistress of Doom](http://www.wowhead.com/npc=19192/mistress-of-doom#abilities) don't despawn, at least in this post-TBC video: https://youtu.be/bGYAasxZHZQ?t=6m37s

**Comment by KimmoKM** [comment]
This also seems to apply to pets/minions and "summons" accompanying the caster in general. For example, [Joana's guide](https://youtu.be/YcEqpVfrE7U?t=1h13m45s) shows voidwalkers of Burning Blade Apprentices staying around after their master has died. A [comment](http://www.wowhead.com/npc=3198/burning-blade-apprentice#comments:id=584119) from WotLK demonstrates this detail hasn't been changed since. I would suspect this might apply to mobs such as [Shadowy Summoners](http://www.wowhead.com/npc=17088/shadowy-summoner#comments) with their succubi or Moonstalker Matriarchs and their Runt "pets" at Darkshore as well, although a quick search didn't yield any videos that would show how these two specific mobs worked.

**Comment by ccnim** [comment]
I can't find proof for either direction, but it seems extremely weird for me that Moonstalker Runts, while displayed as pets for Moonstalker Matriarchs (there are and should be lonesome runts as well), despawn immediately as their mother is killed.

**Comment by KimmoKM** [comment]
Arcane Fiends summoned by [Ethereal Sorcerers](http://www.wowhead.com/npc=18313/ethereal-sorcerer#comments) and [Ethereal Spellbinders](http://www.wowhead.com/npc=18312/ethereal-spellbinder#comments) despawn immediately after being summoned. I couldn't find direct evidence on how the mechanic is supposed to work, but seeing as that comments mention the mobs being tough because of the adds, it would seem weird if they are alive for one second and then immediately despawn. I'm not going to make a whole new issue because this is largely speculation (you'd probably have to test this in retail), but this is related: comments also mention that the adds spawn after set amount of time while in CC they spawn every time even if the mob is being killed in seconds after CC finishes. Would make sense, considering I didn't see the adds being spawned once in the videos I went through (mostly timewalking from retail, but a couple of runs from retail TBC as well, in which case the mobs didn't go down nearly as fast as in our run).
